Automattic/newspack-plugin
# Feat/wizard permissions

### All Submissions:






### Changes proposed in this Pull Request:



Makes the access checks for the wizards more versatile, not single-capability-based. This will allow access to various areas of the product for different users. 

### How to test the changes in this Pull Request:

1. Set `newspack-newsletters` to `feat/capabilities` branch (https://github.com/Automattic/newspack-newsletters/pull/1398)
2. Set `newspack-popups` to `feat/capabilities` branch (https://github.com/Automattic/newspack-popups/pull/1259)
1. As a site administrator, verify that you can see all the expected items in the Newspack Dashboard and the the wizards work as expected*
1. Install the [capability-manager-enhanced plugin](https://wordpress.org/plugins/capability-manager-enhanced/)
3. Create a new role in Capabilities -> Roles and then open Capabilities -> Capabilities. Choose the new role in the dropdown
4. Assign Posts -> Edit in "Editing" group and "read" in "Admin" group – this will let the user with the role access /wp-admin
5. Open the "Newspack" group and observe a list of capabilities – one for each wizard** 
6. Assign the `newspack wizard newspack-syndication-wizard` to the role and save it
7. Create a new user and assign the newly created role to them
8. Log in as the new user and observe the Newspack menu item feature the single chosen wizard, and likewise the Newspack Dashboard
9. One by one, test access to all wizards
    1. For Newsletters and Campaigns, apart from the wizard cap you'll also have to assign all caps present in the "Newspack Newsletters" and "Newspack Campaigns" groups

\* I know, that's a lot to ask. But frankly these changes touch everything about the wizards 😬
\** Note that `setup`, `settings`, and `components-demo` wizards won't appear on the dashboard. These are special wizards, not meant for access from dashboard:

- Setup wizard is used for a first-time Newspack setup
- Settings wizard does not have a UI, but it might at some point
- Components Demo is a showcase/development-help for the UI used in the Newspack admin areas. It can be accessed directly at `/wp-admin/admin.php?page=newspack-components-demo` or via a wizard footer link if `WP_NEWSPACK_DEBUG` env variable is set. 

### Other information:
